src/functionaltest/java/com/ericsson/ei/config/SeleniumConfig.java

```diff
@@ -70,32 +67,57 @@ public static String getBaseUrl(int randomServerPort) {
         return "http://localhost:" + randomServerPort;
     }
 
-    private static FirefoxBinary installFirefoxBinary() {
-        String firefoxBZip2FileNameLinux = FilenameUtils.getName(firefoxBZip2FileUrlLinux);
-
-        String firefoxBZip2FilePath = String.join(
-                File.separator, tempDownloadDirectory.getPath(), firefoxBZip2FileNameLinux);
-        Utils.downloadFileFromUrlToDestination(firefoxBZip2FileUrlLinux, firefoxBZip2FilePath);
-        Utils.extractBZip2InDir(firefoxBZip2FilePath, tempDownloadDirectory.getPath());
+    private static FirefoxBinary getFirefoxBinary() {
+        String firefoxPath = getFirefoxDirPath().getPath();
         File firefoxBinaryFilePath = new File(
-                String.join(File.separator, tempDownloadDirectory.getPath(), "firefox", "firefox"));
+                String.join(File.separator, firefoxPath, "firefox", "firefox"));
+
+        if (firefoxBinaryFilePath.isFile()) {
+            LOGGER.debug("Reusing existing firefox binary.");
+            return new FirefoxBinary(firefoxBinaryFilePath);
+        }
+
+        LOGGER.debug("Downloading and extracting new Firefox binary.");
+        final String firefoxTarFileUrl = getFirefoxTarFileUrl();
+
+        String firefoxBZip2FileNameLinux = FilenameUtils.getName(firefoxTarFileUrl);
+        String firefoxTarFilePath = String.join(
+                File.separator, firefoxPath, firefoxBZip2FileNameLinux);
+
+        Utils.downloadFileFromUrlToDestination(firefoxTarFileUrl, firefoxTarFilePath);
+        Utils.extractBZip2InDir(firefoxTarFilePath, firefoxPath);
         Utils.makeBinFileExecutable(firefoxBinaryFilePath);
-        FirefoxBinary firefoxBinary = new FirefoxBinary(firefoxBinaryFilePath);
-        return firefoxBinary;
+
+        return new FirefoxBinary(firefoxBinaryFilePath);
+    }
+
+    private static File getFirefoxDirPath() {
+        String relPath = SeleniumConfig.class.getProtectionDomain()
```

Comment:
Should we add a comment here saying that we are looking the in target directory for the binary?